lib/motsu-proc/src/test.rs

```diff
@@ -21,31 +21,49 @@ pub(crate) fn test(_attr: &TokenStream, input: TokenStream) -> TokenStream {
     }
 
     // Whether 1 or none contracts will be declared.
-    let contract_declarations = fn_args.into_iter().map(|arg| {
-        let FnArg::Typed(arg) = arg else {
-            error!(arg, "unexpected receiver argument in test signature");
-        };
-        let contract_arg_binding = &arg.pat;
-        let contract_ty = &arg.ty;
+    let arg_binding_and_ty = match fn_args
+        .into_iter()
+        .map(|arg| {
+            let FnArg::Typed(arg) = arg else {
+                error!(@arg, "unexpected receiver argument in test signature");
+            };
+            let contract_arg_binding = &arg.pat;
+            let contract_ty = &arg.ty;
+            Ok((contract_arg_binding, contract_ty))
+        })
+        .collect::<Result<Vec<_>, _>>()
+    {
+        Ok(res) => res,
+        Err(err) => return err.to_compile_error().into(),
+    };
 
-        // Test case assumes, that contract's variable has `&mut` reference
-        // to contract's type.
-        quote! {
-            let mut #contract_arg_binding = <#contract_ty>::default();
-            let #contract_arg_binding = &mut #contract_arg_binding;
-        }
-    });
+    let contract_arg_defs =
+        arg_binding_and_ty.iter().map(|(arg_binding, contract_ty)| {
+            // Test case assumes, that contract's variable has `&mut` reference
+            // to contract's type.
+            quote! {
+                #arg_binding: &mut #contract_ty
+            }
+        });
+
+    let contract_args =
+        arg_binding_and_ty.iter().map(|(_arg_binding, contract_ty)| {
+            // Pass mutable reference to the contract.
+            quote! {
+                &mut <#contract_ty>::default()
+            }
+        });
 
-    // Output full testcase function.
-    // Declare contract.
-    // And in the end, reset storage for test context.
+    // Declare test case closure.
+    // Pass mut ref to the test closure and call it.
+    // Reset storage for the test context and return test's output.
     quote! {
         #( #attrs )*
         #[test]
         fn #fn_name() #fn_return_type {
             use ::motsu::prelude::DefaultStorage;
-            #( #contract_declarations )*
-            let res = #fn_block;
+            let test = | #( #contract_arg_defs ),* | #fn_block;
+            let res = test( #( #contract_args ),* );
             ::motsu::prelude::Context::current().reset_storage();
             res
         }
```

Comment:
In the PR description you mention that you added missing tests to code coverage, but there aren't any here. 
Maybe you forgot to push?

Reply:
That is a very good question:D
Actually these few lines of code, where you commented did main job..
Each test case body, even marked with `#[motsu::test]` is included into code coverage now.
By the end we've received more covered code lines, that inflated total code coverage by 11-12%.
A bit scammy, but I feel there is more value in test cases and test functions included into coverage. And target for a higher percentage of coverage by the end.

Reply:
I really don't see the benefit of including _tests_ in code coverage. The primary concern of code coverage metrics is ensuring the reliability and correctness of _the production code_, or in our case the contracts. 

This is why `cargo-llvm-cov`, our coverage tool, [excludes test files by default](https://github.com/taiki-e/cargo-llvm-cov#exclude-function-from-coverage).

Making sure that test lines are covered serves no practical purpose other than inflating the coverage percentage.